### DISCUSSION TITLE
[Developer Tooling] Truffle & Ganache

### DISCUSSION THREAD

**Discussion by Lizdaldal** [filecoin-project/community#92] [Project Showcase]
Hey All! Starting this discussion for updates on[ Truffle Preserve](https://github.com/filecoin-project/community/blob/master/projects/developer_tooling/truffle_preserve.md) and [Ganache Flavored Filecoin](https://github.com/filecoin-project/community/blob/master/projects/developer_tooling/filecoin-flavored-ganache.md). We have alpha releases of both and we are currently working on stable releases. We'll post regular updates for you here & would love your feedback.

**Comment by Lizdaldal** [comment]
Hey All! Below is an update for the week of March 1, 2021. We'll also have a demo video coming your way next week! 

- Majority of Ganache code has initial code reviews
- Continuing to mature Ganache implementations and fixing bugs
- Added configurable storage deal options to preserve-to-filecoin
- Improved error handling for preserve-to-filecoin
- Expanded tests for preserve-to-filecoin
- Started on implementing preserve-to-buckets

**Comment by Lizdaldal** [comment]
Hello!

We are just days away from the Truffle and Ganache for Filecoin release. Below are some updates on our status:

- Complete @truffle/preserve-to-buckets
- Prioritise nice-to-haves
- Refactor truffle-preserve recipe dependency system
- Majority of Ganache code reviews are complete and merged
- Gearing up to do a practice release of Ganache UI
- Finalizing Ganache documentation

**Comment by Lizdaldal** [comment]
It's the final countdown.. we will have Ganache and Truffle released for you next week!

- Finished refactoring recipe dependency system
- Published a pre-release for truffle@preserves
- Almost all Ganache PRs merged
- Finished Ganache documentation and work in progress on Truffle documentation
- Ganache release candidate version being made for final testing 
- Working with Protocol Labs team on launch plan

**Reply by jennijuju** [reply]
@Lizdaldal Thank you for the upgrade! Can we feature your latest updates in a showcase video?

**Reply by Lizdaldal** [reply]
Yes! 😊

**Comment by Lizdaldal** [comment]
Hello all!

We are excited to share Truffle and Ganache for Filecoin are officially available! 🎊

You can check out our training video on how to get started here 👉https://youtu.be/gW4piWkTvJw

- Filecoin-flavored Ganache: https://www.trufflesuite.com/ganache
- truffle preserve recipes/integrations: https://github.com/trufflesuite/truffle/releases/tag/v5.3.0
- Documentation for installing Filecoin-flavored Ganache: https://www.trufflesuite.com/docs/filecoin/ganache/overview
- Filecoin Truffle Box: https://www.trufflesuite.com/boxes
- Our blog post about the Filecoin integrations: https://www.trufflesuite.com/blog/truffle-and-ganache-now-come-in-filecoin-flavor

**Reply by jennijuju** [reply]
Congrats and I'm super excited to try it out soon!
